projects/canopy/src/lib/forms/sort-code/sort-code.component.ts

```diff
@@ -158,18 +158,38 @@ export class LgSortCodeComponent implements OnInit, ControlValueAccessor {
   }
 
   validate(): ValidationErrors {
-    const errors: ValidationErrors = {};
+    const invalidFields: Array<string> = [];
     Object.keys(this.sortCodeFormGroup.controls).forEach((fieldName) => {
       if (
         this.sortCodeFormGroup.controls[fieldName].invalid &&
         (this.sortCodeFormGroup.controls[fieldName].dirty ||
           (this.formGroupDirective && this.formGroupDirective.submitted))
       ) {
-        errors.invalidField = true;
+        invalidFields.push(fieldName);
       }
     });
-    if (Object.keys(errors).length) {
-      return errors;
+
+    const invalidFieldNames: Array<string> = [];
+    const requiredFieldNames: Array<string> = [];
+    invalidFields.forEach((fieldName) => {
```

Comment:
Just an idea, you could replace the foreachs/`Array.push` with `Array.filter`. Just a preference of mine, not sure if that is just me though.

Reply:
I nearly made the same comment, but I figured it would be two loops instead of one. Happy with either from my side.

Reply:
Yeah I'm not too precious about it. Happy to leave as is

Reply:
I've refactored the code to use a single filter.